### PR TITLE
syncer: add test for upstream instance restart

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
#847 in 2.0

### What is changed and how it works?
add failpoint to mock upstream instance restart

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by lance6716** [comment]
LGTM

File: syncer/streamer_controller.go [hunk 1]
```diff
@@ -222,6 +222,25 @@ func (c *StreamerController) GetEvent(tctx *tcontext.Context) (event *replicatio
...
+		if err2 != nil {
+			return
+		}
+		// mock upstream instance restart from 5-20s
+		startTime := t.Add(5 * time.Second)
+		endTime := startTime.Add(20 * time.Second)
+		now := time.Now()
+		if now.Before(endTime) && now.After(startTime) {
+			tctx.L().Info("mock upstream instance restart", zap.String("failpoint", "SyncerGetEventError"), zap.Stringer("startTime", startTime), zap.Stringer("endTime", endTime))
+			failpoint.Return(nil, terror.ErrDBBadConn.Generate())
```
Comment (csuzhangxc) [inline syncer/streamer_controller.go:240]: just one question, in real upstream instance restart, is `terror.ErrDBBadConn` we got?

Comment (GMHDBJD) [reply syncer/streamer_controller.go:240]: It's uncertain. May be it's binlog read error, badconnection error or database driver error

Comment (csuzhangxc) [reply syncer/streamer_controller.go:240]: but we auto-resume for all of them, right?

Comment (GMHDBJD) [reply syncer/streamer_controller.go:240]: Yes, I have manually test it.

**Pull request merged**